server/modules/opensubtitles/opensubtitles.js

```diff
@@ -83,73 +84,62 @@ define(function () {
-                                [SubActualCD] => 1
-                                [SubSize] => 15019
-                                [SubHash] => 0cb51bf4a5266a9aee42a2d8c7ab6793
-                                [IDSubtitle] => 905
-                                [UserID] => 0
-                                [SubLanguageID] => spa
-                                [SubFormat] => srt
-                                [SubSumCD] => 1
-                                [SubAuthorComment] =>
-                                [SubAddDate] => 2005-06-15 20:05:35
-                                [SubBad] => 1
-                                [SubRating] => 4.5
-                                [SubDownloadsCnt] => 216
-                                [MovieReleaseName] => ss
-                                [IDMovie] => 11517
-                                [IDMovieImdb] => 349076
-                                [MovieName] => Al sur de Granada
-                                [MovieNameEng] => South from Granada
-                                [MovieYear] => 2003
-                                [MovieImdbRating] => 6.4
-                                [SubFeatured] => 0
-                                [UserNickName] =>
-                                [ISO639] => es
-                                [LanguageName] => Spanish
-                                [SubComments] => 1
-                                [SubHearingImpaired] => 0
-                                [UserRank] =>
-                                [SeriesSeason] =>
-                                [SeriesEpisode] =>
-                                [MovieKind] => movie
-                                [SubDownloadLink] => http://dl.opensubtitles.org/en/download/filead/1118.gz
-                                [ZipDownloadLink] => http://dl.opensubtitles.org/en/download/subad/905
-                                [SubtitlesLink] => http://www.opensubtitles.org/en/subtitles/905/al-sur-de-granada-es
-                            )
-
-                        [1] => Array
-           */
+    };
+
+    OpenSubtitles.DownloadAll = function (_options, _callback) {
+        const path = require('path');
+        const os = require('os');
+        const async = require('async');
+        const archiver = require('archiver');
+        fs.mkdtemp(os.tmpdir() + path.sep + 'sub-', (err, folder) => {
+            if (err) throw err;
+
+            OpenSubtitles.SearchSubtitles(_options.movie, _options.language, function (subtitles) {
+                var filteredSubs = subtitles.filter(function (_subtitle) {
+                    return _subtitle.SubFormat === 'srt';
+                });
+
+                async.each(filteredSubs, function (file, callback) {
+                    OpenSubtitles.DownloadSubtitles(folder, file, callback);
+                }, function (err) {
+                    if (err) {
+                        throw err;
```

Comment:
is this like Java where the throw finishes the execution? If yes, we could remove the else statement to avoid a tab space, a new scope, and curly braces to manage.